qiskit/providers/models/jobstatus.py

```diff
@@ -12,40 +12,58 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""Model and schema for job status."""
+"""Class for job status."""
 
-from marshmallow.validate import OneOf
+from types import SimpleNamespace
+from qiskit.exceptions import QiskitError
 
-from qiskit.validation import BaseModel, BaseSchema, bind_schema
-from qiskit.validation.fields import String
 
-
-class JobStatusSchema(BaseSchema):
-    """Schema for JobStatus."""
-
-    # Required properties.
-    job_id = String(required=True)
-    status = String(required=True,
-                    validate=OneOf(['DONE', 'QUEUED', 'CANCELLED', 'RUNNING', 'ERROR']))
-    status_msg = String(required=True)
-
-
-@bind_schema(JobStatusSchema)
-class JobStatus(BaseModel):
+class JobStatus(SimpleNamespace):
     """Model for JobStatus.
 
-    Please note that this class only describes the required fields. For the
-    full description of the model, please check ``JobStatusSchema``.
-
     Attributes:
         job_id (str): backend job_id.
         status (str): status of the job.
         status_msg (str): status message.
     """
+    job_states = ['DONE', 'QUEUED', 'CANCELLED', 'RUNNING', 'ERROR']
 
     def __init__(self, job_id, status, status_msg, **kwargs):
         self.job_id = job_id
+        if status not in self.job_states:
```

Comment:
This is probably outside the scope of this PR, but do we really want to restrict the job status to those 5? Especially since there are more statuses than those defined in `qiskit/providers/jobstatus`.

Reply:
I will remove this restriction, I was just mirroring what was in the schema definition. But, yeah it does seem like an unnecessary restriction. That being said the subclasses will all likely overload this.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/4016/commits/89590797d2326383a3bdba812518fe2466c5556d